In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

pd.set_option("display.max_columns",None)
pd.set_option("display.max_rows",None)
df=pd.read_csv("application_data.csv")

In [ ]:
#knowing the data
print("top 5 rows of data:\n\n",df.head(5))

top 5 rows of data:

    SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0      100002       1         Cash loans           M            N   
1      100003       0         Cash loans           F            N   
2      100004       0    Revolving loans           M            Y   
3      100006       0         Cash loans           F            N   
4      100007       0         Cash loans           M            N   

  FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0               Y             0          202500.0    406597.5      24700.5   
1               N             0          270000.0   1293502.5      35698.5   
2               Y             0           67500.0    135000.0       6750.0   
3               Y             0          135000.0    312682.5      29686.5   
4               Y             0          121500.0    513000.0      21865.5   

   AMT_GOODS_PRICE NAME_TYPE_SUITE NAME_INCOME_TYPE  \
0         351000.0   Unaccompanied     

In [ ]:
print("Getting information about column\n\n", df.dtypes)

Getting information about column

 SK_ID_CURR                        int64
TARGET                            int64
NAME_CONTRACT_TYPE               object
CODE_GENDER                      object
FLAG_OWN_CAR                     object
FLAG_OWN_REALTY                  object
CNT_CHILDREN                      int64
AMT_INCOME_TOTAL                float64
AMT_CREDIT                      float64
AMT_ANNUITY                     float64
AMT_GOODS_PRICE                 float64
NAME_TYPE_SUITE                  object
NAME_INCOME_TYPE                 object
NAME_EDUCATION_TYPE              object
NAME_FAMILY_STATUS               object
NAME_HOUSING_TYPE                object
REGION_POPULATION_RELATIVE      float64
DAYS_BIRTH                        int64
DAYS_EMPLOYED                     int64
DAYS_REGISTRATION               float64
DAYS_ID_PUBLISH                   int64
OWN_CAR_AGE                     float64
FLAG_MOBIL                        int64
FLAG_EMP_PHONE                    int64
FLAG_

In [ ]:
print("Getting name of all columns: \n\n",df.columns)

Getting name of all columns: 

 Index(['SK_ID_CURR', 'TARGET', 'NAME_CONTRACT_TYPE', 'CODE_GENDER',
       'FLAG_OWN_CAR', 'FLAG_OWN_REALTY', 'CNT_CHILDREN', 'AMT_INCOME_TOTAL',
       'AMT_CREDIT', 'AMT_ANNUITY',
       ...
       'FLAG_DOCUMENT_18', 'FLAG_DOCUMENT_19', 'FLAG_DOCUMENT_20',
       'FLAG_DOCUMENT_21', 'AMT_REQ_CREDIT_BUREAU_HOUR',
       'AMT_REQ_CREDIT_BUREAU_DAY', 'AMT_REQ_CREDIT_BUREAU_WEEK',
       'AMT_REQ_CREDIT_BUREAU_MON', 'AMT_REQ_CREDIT_BUREAU_QRT',
       'AMT_REQ_CREDIT_BUREAU_YEAR'],
      dtype='object', length=122)


In [ ]:
print(f"no. of row:{df.shape[0]}\n no. of columns: {df.shape[1]}")

no. of row:307511
 no. of columns: 122


In [ ]:
#checking for nan values
(df.isna().sum()/len(df)*100).sort_values(axis=0)   

SK_ID_CURR                       0.000000
HOUR_APPR_PROCESS_START          0.000000
REG_REGION_NOT_WORK_REGION       0.000000
LIVE_REGION_NOT_WORK_REGION      0.000000
REG_CITY_NOT_LIVE_CITY           0.000000
REG_CITY_NOT_WORK_CITY           0.000000
LIVE_CITY_NOT_WORK_CITY          0.000000
ORGANIZATION_TYPE                0.000000
FLAG_DOCUMENT_21                 0.000000
FLAG_DOCUMENT_20                 0.000000
FLAG_DOCUMENT_19                 0.000000
FLAG_DOCUMENT_18                 0.000000
FLAG_DOCUMENT_17                 0.000000
FLAG_DOCUMENT_16                 0.000000
FLAG_DOCUMENT_15                 0.000000
FLAG_DOCUMENT_14                 0.000000
FLAG_DOCUMENT_13                 0.000000
FLAG_DOCUMENT_12                 0.000000
FLAG_DOCUMENT_11                 0.000000
FLAG_DOCUMENT_10                 0.000000
FLAG_DOCUMENT_9                  0.000000
FLAG_DOCUMENT_8                  0.000000
FLAG_DOCUMENT_7                  0.000000
FLAG_DOCUMENT_6                  0

In [ ]:
print("columns with more than 50% of Nan values \n\n")
na_val=((df.isna().sum()/len(df))*100).sort_values()>50
print("count of columns:",na_val[na_val.values==True].count(),"\n\n ")
print(na_val[na_val.values==True])

columns with more than 50% of Nan values 


count of columns: 41 

 
HOUSETYPE_MODE              True
LIVINGAREA_AVG              True
LIVINGAREA_MODE             True
LIVINGAREA_MEDI             True
ENTRANCES_AVG               True
ENTRANCES_MODE              True
ENTRANCES_MEDI              True
APARTMENTS_MEDI             True
APARTMENTS_AVG              True
APARTMENTS_MODE             True
WALLSMATERIAL_MODE          True
ELEVATORS_MEDI              True
ELEVATORS_AVG               True
ELEVATORS_MODE              True
NONLIVINGAREA_MODE          True
NONLIVINGAREA_AVG           True
NONLIVINGAREA_MEDI          True
EXT_SOURCE_1                True
BASEMENTAREA_MODE           True
BASEMENTAREA_AVG            True
BASEMENTAREA_MEDI           True
LANDAREA_MEDI               True
LANDAREA_AVG                True
LANDAREA_MODE               True
OWN_CAR_AGE                 True
YEARS_BUILD_MODE            True
YEARS_BUILD_AVG             True
YEARS_BUILD_MEDI            True
FLOORSM

In [ ]:
df.shape

(307511, 122)

In [ ]:
df.describe().transpose()
#	Target variable 
#1 - client with payment difficulties:he/she had late payment more than X days on at least one of the first Y installments of the loan in our sample,
#0 - all other cases

count           mean            std  \
SK_ID_CURR                    307511.0  278180.518577  102790.175348   
TARGET                        307511.0       0.080729       0.272419   
CNT_CHILDREN                  307511.0       0.417052       0.722121   
AMT_INCOME_TOTAL              307511.0  168797.919297  237123.146279   
AMT_CREDIT                    307511.0  599025.999706  402490.776996   
AMT_ANNUITY                   307499.0   27108.573909   14493.737315   
AMT_GOODS_PRICE               307233.0  538396.207429  369446.460540   
REGION_POPULATION_RELATIVE    307511.0       0.020868       0.013831   
DAYS_BIRTH                    307511.0  -16036.995067    4363.988632   
DAYS_EMPLOYED                 307511.0   63815.045904  141275.766519   
DAYS_REGISTRATION             307511.0   -4986.120328    3522.886321   
DAYS_ID_PUBLISH               307511.0   -2994.202373    1509.450419   
OWN_CAR_AGE                   104582.0      12.061091      11.944812   
FLAG_MOBIL                    307511.0       0.999997       0.001803   
FLAG_EMP_PHONE                307511.0       0.819889       0.384280   
FLAG_WORK_PHONE               307511.0       0.199368       0.399526   
FLAG_CONT_MOBILE              307511.0       0.998133       0.043164   
FLAG_PHONE                    307511.0       0.281066       0.449521   
FLAG_EMAIL                    307511.0       0.056720       0.231307   
CNT_FAM_MEMBERS               307509.0       2.152665       0.910682   
REGION_RATING_CLIENT          307511.0       2.052463       0.509034   
REGION_RATING_CLIENT_W_CITY   307511.0       2.031521       0.502737   
HOUR_APPR_PROCESS_START       307511.0      12.063419       3.265832   
REG_REGION_NOT_LIVE_REGION    307511.0       0.015144       0.122126   
REG_REGION_NOT_WORK_REGION    307511.0       0.050769       0.219526   
LIVE_REGION_NOT_WORK_REGION   307511.0       0.040659       0.197499   
REG_CITY_NOT_LIVE_CITY        307511.0       0.078173       0.268444   
REG_CITY_NOT_WORK_CITY        307511.0       0.230454       0.421124   
LIVE_CITY_NOT_WORK_CITY       307511.0       0.179555       0.383817   
EXT_SOURCE_1                  134133.0       0.502130       0.211062   
EXT_SOURCE_2                  306851.0       0.514393       0.191060   
EXT_SOURCE_3                  246546.0       0.510853       0.194844   
APARTMENTS_AVG                151450.0       0.117440       0.108240   
BASEMENTAREA_AVG              127568.0       0.088442       0.082438   
YEARS_BEGINEXPLUATATION_AVG   157504.0       0.977735       0.059223   
YEARS_BUILD_AVG               103023.0       0.752471       0.113280   
COMMONAREA_AVG                 92646.0       0.044621       0.076036   
ELEVATORS_AVG                 143620.0       0.078942       0.134576   
ENTRANCES_AVG                 152683.0       0.149725       0.100049   
FLOORSMAX_AVG                 154491.0       0.226282       0.144641   
FLOORSMIN_AVG                  98869.0       0.231894       0.161380   
LANDAREA_AVG                  124921.0       0.066333       0.081184   
LIVINGAPARTMENTS_AVG           97312.0       0.100775       0.092576   
LIVINGAREA_AVG                153161.0       0.107399       0.110565   
NONLIVINGAPARTMENTS_AVG        93997.0       0.008809       0.047732   
NONLIVINGAREA_AVG             137829.0       0.028358       0.069523   
APARTMENTS_MODE               151450.0       0.114231       0.107936   
BASEMENTAREA_MODE             127568.0       0.087543       0.084307   
YEARS_BEGINEXPLUATATION_MODE  157504.0       0.977065       0.064575   
YEARS_BUILD_MODE              103023.0       0.759637       0.110111   
COMMONAREA_MODE                92646.0       0.042553       0.074445   
ELEVATORS_MODE                143620.0       0.074490       0.132256   
ENTRANCES_MODE                152683.0       0.145193       0.100977   
FLOORSMAX_MODE                154491.0       0.222315       0.143709   
FLOORSMIN_MODE                 98869.0       0.228058       0.161160   
L

# SUMMARY

1. we have 307511 row and  122 columns

2. 41 columns in dataset are having more than 50% of nan values

3. 8% are the people with payment issue

4. Almost all columns have less standard deviation which means low variance means nearer to mean

5. The data seem to be disbalanced with respect to target variable

if the standard deviation is high that means your data have high variance. And if you compare the mean and 50% percentile (median) you can see the skewness of your numeric variables. From skewness you can infer that your numeric variables have too many outlier or not.


standard deviation----->variance

mean,mode---->skewness

sknewness--->outliers


positively skewed:   mean of the data is greater than the median 

mean=median= mode symmetrical distribution

negatively skewed: mean is less than median


we find skewness of data so that we can transform data in (normal distribution)

In [ ]:
df["NAME_CONTRACT_TYPE"].value_counts()

Cash loans         278232
Revolving loans     29279
Name: NAME_CONTRACT_TYPE, dtype: int64

we have 36935 client with cash loan and 3847 with revolving loans

In [ ]:
df["TARGET"].value_counts()

0    282686
1     24825
Name: TARGET, dtype: int64

In [ ]:
3299/len(df)*100

1.0728071516140885

8% i.e 3299 in the records have payment issue


In [ ]:
df["CODE_GENDER"].value_counts()

F      202448
M      105059
XNA         4
Name: CODE_GENDER, dtype: int64

We have 2:1 ratio in our records for F:M 

In [ ]:
df_info=df[["AMT_INCOME_TOTAL","AMT_CREDIT"]]
df_info["times"]=np.floor(df["AMT_CREDIT"]/df["AMT_INCOME_TOTAL"])
df_info.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


AMT_INCOME_TOTAL  AMT_CREDIT  times
0          202500.0    406597.5    2.0
1          270000.0   1293502.5    4.0
2           67500.0    135000.0    2.0
3          135000.0    312682.5    2.0
4          121500.0    513000.0    4.0

In [ ]:
#Education based defaluters
df[df["TARGET"]==1]["NAME_EDUCATION_TYPE"].value_counts()

Secondary / secondary special    19524
Higher education                  4009
Incomplete higher                  872
Lower secondary                    417
Academic degree                      3
Name: NAME_EDUCATION_TYPE, dtype: int64

In [ ]:
#Income type defaluters
df[df["TARGET"]==1]["NAME_INCOME_TYPE"].value_counts()

Working                 15224
Commercial associate     5360
Pensioner                2982
State servant            1249
Unemployed                  8
Maternity leave             2
Name: NAME_INCOME_TYPE, dtype: int64

# Summary



**To see the distribution of contract type**     

      Contract type       Frequency
      Cash loans           36935
      Revolving loans      3847


**8% i.e 3299 in the records have payment issue**

**female to male ratio is 2:1**

F:      202448

M:      105059

**To see income and credit ratio**

	AMT_INCOME_TOTAL	AMT_CREDIT	times

	202500.0	        406597.5	   2.0

	270000.0	        1293502.5	   4.0

	67500.0	          135000.0	   2.0

	135000.0	        312682.5	   2.0

	121500.0	        513000.0	   4.0



**Education based defaluters**

Secondary / secondary special    19524

Higher education                  4009

Incomplete higher                  872

Lower secondary                    417

Academic degree                      3


**Income type defaluters**

Working                 15224

Commercial associate     5360

Pensioner                2982

State servant            1249

Unemployed                  8

Maternity leave             2

In [ ]:
#Dividing the columns into continuous and discreate column
#threshold =30
cont_col=[]
dis_col=[]
for column in df.columns:
  if len(df[column].unique())>30:
    cont_col.append(column)
  else:
    dis_col.append(column)

#removing the target variable from dis_col
dis_col.remove("TARGET")

In [ ]:
print(f"No of continuous columns:{len(cont_col)} \n No. of discrete columns {len(dis_col)}")

No of continuous columns:58 
 No. of discrete columns 63


In [ ]:
df_p=pd.read_csv("/content/gdrive/MyDrive/Colab Notebooks/previous_application.csv")
df_p.head(5)

SK_ID_PREV  SK_ID_CURR NAME_CONTRACT_TYPE  AMT_ANNUITY  AMT_APPLICATION  \
0     2030495      271877     Consumer loans     1730.430          17145.0   
1     2802425      108129         Cash loans    25188.615         607500.0   
2     2523466      122040         Cash loans    15060.735         112500.0   
3     2819243      176158         Cash loans    47041.335         450000.0   
4     1784265      202054         Cash loans    31924.395         337500.0   

   AMT_CREDIT  AMT_DOWN_PAYMENT  AMT_GOODS_PRICE WEEKDAY_APPR_PROCESS_START  \
0     17145.0               0.0          17145.0                   SATURDAY   
1    679671.0               NaN         607500.0                   THURSDAY   
2    136444.5               NaN         112500.0                    TUESDAY   
3    470790.0               NaN         450000.0                     MONDAY   
4    404055.0               NaN         337500.0                   THURSDAY   

   HOUR_APPR_PROCESS_START FLAG_LAST_APPL_PER_CONTRACT  \
0                       15                           Y   
1                       11                           Y   
2                       11                           Y   
3                        7                           Y   
4                        9                           Y   

   NFLAG_LAST_APPL_IN_DAY  RATE_DOWN_PAYMENT  RATE_INTEREST_PRIMARY  \
0                       1                0.0               0.182832   
1                       1                NaN                    NaN   
2                       1                NaN                    NaN   
3                       1                NaN                    NaN   
4                       1                NaN                    NaN   

   RATE_INTEREST_PRIVILEGED NAME_CASH_LOAN_PURPOSE NAME_CONTRACT_STATUS  \
0                  0.867336                    XAP             Approved   
1                       NaN                    XNA             Approved   
2                       NaN                    XNA             Approved   
3                       NaN                    XNA             Approved   
4                       NaN                Repairs              Refused   

   DAYS_DECISION      NAME_PAYMENT_TYPE CODE_REJECT_REASON  NAME_TYPE_SUITE  \
0            -73  Cash through the bank                XAP              NaN   
1           -164                    XNA                XAP    Unaccompanied   
2           -301  Cash through the bank                XAP  Spouse, partner   
3           -512  Cash through the bank                XAP              NaN   
4           -781  Cash through the bank                 HC              NaN   

  NAME_CLIENT_TYPE NAME_GOODS_CATEGORY NAME_PORTFOLIO NAME_PRODUCT_TYPE  \
0         Repeater              Mobile            POS               XNA   
1         Repeater                 XNA           Cash            x-sell   
2         Repeater                 XNA           Cash            x-sell   
3         Repeater                 XNA           Cash            x-sell   
4         Repeater                 XNA           Cash           walk-in   

              CHANNEL_TYPE  SELLERPLACE_AREA NAME_SELLER_INDUSTRY  \
0             Country-wide                35         Connectivity   
1           Contact center                -1                  XNA   
2  Credit and cash offices                -1                  XNA   
3  Credit and cash offices                -1                  XNA   
4  Credit and cash offices                -1                  XNA   

   CNT_PAYMENT NAME_YIELD_GROUP       PRODUCT_COMBINATION  DAYS_FIRST_DRAWING  \
0         12.0           middle  POS mobile with interest            365243.0   
1         36.0       low_action          Cash X-Sell: low            365243.0   
2         12.0             high         Cash X-Sell: high            365243.0   
3         12.0           middle       Cash X-Sell: middle            365243.0   
4         24.0             high         Cash Street: high                 NaN   

   D

In [ ]:
print(f"No of rows and columns are:{df_p.shape}" )

No of rows and columns are:(1670214, 37)


In [ ]:
print("Getting information about column\n\n", df_p.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1670214 entries, 0 to 1670213
Data columns (total 37 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   SK_ID_PREV                   1670214 non-null  int64  
 1   SK_ID_CURR                   1670214 non-null  int64  
 2   NAME_CONTRACT_TYPE           1670214 non-null  object 
 3   AMT_ANNUITY                  1297979 non-null  float64
 4   AMT_APPLICATION              1670214 non-null  float64
 5   AMT_CREDIT                   1670213 non-null  float64
 6   AMT_DOWN_PAYMENT             774370 non-null   float64
 7   AMT_GOODS_PRICE              1284699 non-null  float64
 8   WEEKDAY_APPR_PROCESS_START   1670214 non-null  object 
 9   HOUR_APPR_PROCESS_START      1670214 non-null  int64  
 10  FLAG_LAST_APPL_PER_CONTRACT  1670214 non-null  object 
 11  NFLAG_LAST_APPL_IN_DAY       1670214 non-null  int64  
 12  RATE_DOWN_PAYMENT            774370 non-nu

In [ ]:
print("Getting name of all columns: \n\n",df_p.columns)

Getting name of all columns: 

 Index(['SK_ID_PREV', 'SK_ID_CURR', 'NAME_CONTRACT_TYPE', 'AMT_ANNUITY',
       'AMT_APPLICATION', 'AMT_CREDIT', 'AMT_DOWN_PAYMENT', 'AMT_GOODS_PRICE',
       'WEEKDAY_APPR_PROCESS_START', 'HOUR_APPR_PROCESS_START',
       'FLAG_LAST_APPL_PER_CONTRACT', 'NFLAG_LAST_APPL_IN_DAY',
       'RATE_DOWN_PAYMENT', 'RATE_INTEREST_PRIMARY',
       'RATE_INTEREST_PRIVILEGED', 'NAME_CASH_LOAN_PURPOSE',
       'NAME_CONTRACT_STATUS', 'DAYS_DECISION', 'NAME_PAYMENT_TYPE',
       'CODE_REJECT_REASON', 'NAME_TYPE_SUITE', 'NAME_CLIENT_TYPE',
       'NAME_GOODS_CATEGORY', 'NAME_PORTFOLIO', 'NAME_PRODUCT_TYPE',
       'CHANNEL_TYPE', 'SELLERPLACE_AREA', 'NAME_SELLER_INDUSTRY',
       'CNT_PAYMENT', 'NAME_YIELD_GROUP', 'PRODUCT_COMBINATION',
       'DAYS_FIRST_DRAWING', 'DAYS_FIRST_DUE', 'DAYS_LAST_DUE_1ST_VERSION',
       'DAYS_LAST_DUE', 'DAYS_TERMINATION', 'NFLAG_INSURED_ON_APPROVAL'],
      dtype='object')


In [ ]:
df_p.describe().transpose()

count          mean            std  \
SK_ID_PREV                 1670214.0  1.923089e+06  532597.958696   
SK_ID_CURR                 1670214.0  2.783572e+05  102814.823849   
AMT_ANNUITY                1297979.0  1.595512e+04   14782.137335   
AMT_APPLICATION            1670214.0  1.752339e+05  292779.762387   
AMT_CREDIT                 1670213.0  1.961140e+05  318574.616546   
AMT_DOWN_PAYMENT            774370.0  6.697402e+03   20921.495410   
AMT_GOODS_PRICE            1284699.0  2.278473e+05  315396.557937   
HOUR_APPR_PROCESS_START    1670214.0  1.248418e+01       3.334028   
NFLAG_LAST_APPL_IN_DAY     1670214.0  9.964675e-01       0.059330   
RATE_DOWN_PAYMENT           774370.0  7.963682e-02       0.107823   
RATE_INTEREST_PRIMARY         5951.0  1.883569e-01       0.087671   
RATE_INTEREST_PRIVILEGED      5951.0  7.735025e-01       0.100879   
DAYS_DECISION              1670214.0 -8.806797e+02     779.099667   
SELLERPLACE_AREA           1670214.0  3.139511e+02    7127.443459   
CNT_PAYMENT                1297984.0  1.605408e+01      14.567288   
DAYS_FIRST_DRAWING          997149.0  3.422099e+05   88916.115834   
DAYS_FIRST_DUE              997149.0  1.382627e+04   72444.869708   
DAYS_LAST_DUE_1ST_VERSION   997149.0  3.376777e+04  106857.034789   
DAYS_LAST_DUE               997149.0  7.658240e+04  149647.415123   
DAYS_TERMINATION            997149.0  8.199234e+04  153303.516729   
NFLAG_INSURED_ON_APPROVAL   997149.0  3.325702e-01       0.471134   

                                    min           25%           50%  \
SK_ID_PREV                 1.000001e+06  1.461857e+06  1.923110e+06   
SK_ID_CURR                 1.000010e+05  1.893290e+05  2.787145e+05   
AMT_ANNUITY                0.000000e+00  6.321780e+03  1.125000e+04   
AMT_APPLICATION            0.000000e+00  1.872000e+04  7.104600e+04   
AMT_CREDIT                 0.000000e+00  2.416050e+04  8.054100e+04   
AMT_DOWN_PAYMENT          -9.000000e-01  0.000000e+00  1.638000e+03   
AMT_GOODS_PRICE            0.000000e+00  5.084100e+04  1.123200e+05   
HOUR_APPR_PROCESS_START    0.000000e+00  1.000000e+01  1.200000e+01   
NFLAG_LAST_APPL_IN_DAY     0.000000e+00  1.000000e+00  1.000000e+00   
RATE_DOWN_PAYMENT         -1.497876e-05  0.000000e+00  5.160508e-02   
RATE_INTEREST_PRIMARY      3.478125e-02  1.607163e-01  1.891222e-01   
RATE_INTEREST_PRIVILEGED   3.731501e-01  7.156448e-01  8.350951e-01   
DAYS_DECISION             -2.922000e+03 -1.300000e+03 -5.810000e+02   
SELLERPLACE_AREA          -1.000000e+00 -1.000000e+00  3.000000e+00   
CNT_PAYMENT                0.000000e+00  6.000000e+00  1.200000e+01   
DAYS_FIRST_DRAWING        -2.922000e+03  3.652430e+05  3.652430e+05   
DAYS_FIRST_DUE            -2.892000e+03 -1.628000e+03 -8.310000e+02   
DAYS_LAST_DUE_1ST_VERSION -2.801000e+03 -1.242000e+03 -3.610000e+02   
DAYS_LAST_DUE             -2.889000e+03 -1.314000e+03 -5.370000e+02   
DAYS_TERMINATION          -2.874000e+03 -1.270000e+03 -4.990000e+02   
NFLAG_INSURED_ON_APPROVAL  0.000000e+00  0.000000e+00  0.000000e+00   

                                    75%          max  
SK_ID_PREV                 2.384280e+06  2845382.000  
SK_ID_CURR                 3.675140e+05   456255.000  
AMT_ANNUITY                2.065842e+04   418058.145  
AMT_APPLICATION            1.803600e+05  6905160.000  
AMT_CREDIT                 2.164185e+05  6905160.000  
AMT_DOWN_PAYMENT           7.740000e+03  3060045.000  
AMT_GOODS_PRICE            2.340000e+05  6905160.000  
HOUR_APPR_PROCESS_START    1.500000e+01       23.000  
NFLAG_LAST_APPL_IN_DAY     1.000000e+00        1.000  
RATE_DOWN_PAYMENT          1.089091e-01        1.000  
RATE_INTEREST_PRIMARY      1.933299e-01        1.000  
RATE_INTEREST_PRIVILEGED   8.525370e-01        1.000  
DAYS_DECISION             -2.800000e+02       -1.000  
SELLERPLACE_AREA           8.200000e+01  4000000.000  
CNT_PAYMENT                2.400000e+01       84.000  
DAYS_FIRST_DRAWING         3.652430e+05   365243.000  
DAYS_FIRST_DUE

In [ ]:
(df_p.isna().sum()/len(df_p)*100).sort_values()

SK_ID_PREV                      0.000000
NAME_YIELD_GROUP                0.000000
NAME_SELLER_INDUSTRY            0.000000
SELLERPLACE_AREA                0.000000
CHANNEL_TYPE                    0.000000
NAME_PRODUCT_TYPE               0.000000
NAME_PORTFOLIO                  0.000000
NAME_GOODS_CATEGORY             0.000000
NAME_CLIENT_TYPE                0.000000
CODE_REJECT_REASON              0.000000
DAYS_DECISION                   0.000000
NAME_CONTRACT_STATUS            0.000000
NAME_CASH_LOAN_PURPOSE          0.000000
NAME_PAYMENT_TYPE               0.000000
AMT_APPLICATION                 0.000000
NFLAG_LAST_APPL_IN_DAY          0.000000
FLAG_LAST_APPL_PER_CONTRACT     0.000000
HOUR_APPR_PROCESS_START         0.000000
WEEKDAY_APPR_PROCESS_START      0.000000
NAME_CONTRACT_TYPE              0.000000
SK_ID_CURR                      0.000000
AMT_CREDIT                      0.000060
PRODUCT_COMBINATION             0.020716
CNT_PAYMENT                    22.286366
AMT_ANNUITY     

In [ ]:
#correlation between df(application) and df_p(previous application)
col_app=list(df.columns)
col_prev=list(df_p.columns)
uni_col=[]
[uni_col.append(col) for col  in col_app if col not in uni_col]
[uni_col.append(col) for col in col_prev if col not in col_app]

uni_col=list(set(uni_col))
    
print(f"List of columns unique in application.csv and previous application")

List of columns unique in application.csv and previous application


In [ ]:
#common columns
common=[]
[common.append(col) for col in col_app if col in col_prev]
common

['SK_ID_CURR',
 'NAME_CONTRACT_TYPE',
 'AMT_CREDIT',
 'AMT_ANNUITY',
 'AMT_GOODS_PRICE',
 'NAME_TYPE_SUITE',
 'WEEKDAY_APPR_PROCESS_START',
 'HOUR_APPR_PROCESS_START']

In [ ]:
final_df.head(5)

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0      186234       0    Revolving loans           F            N   
1      195755       0         Cash loans           F            Y   
2      278454       0         Cash loans           M            Y   
3      312735       0    Revolving loans           F            N   

  FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0               Y             0           67500.0    270000.0      13500.0   
1               Y             2          142425.0    450000.0      17140.5   
2               N             0          157500.0    900000.0      26316.0   
3               Y             0          202500.0    450000.0      22500.0   

   AMT_GOODS_PRICE NAME_TYPE_SUITE      NAME_INCOME_TYPE  \
0         270000.0   Unaccompanied             Pensioner   
1         450000.0   Unaccompanied  Commercial associate   
2         900000.0   Unaccompanied             Pensioner   
3         450000.0   Unaccompanied             Pensioner   

             NAME_EDUCATION_TYPE NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  \
0  Secondary / secondary special            Married  House / apartment   
1               Higher education            Married  House / apartment   
2               Higher education            Married  House / apartment   
3  Secondary / secondary special              Widow  House / apartment   

   REGION_POPULATION_RELATIVE  DAYS_BIRTH  DAYS_EMPLOYED  DAYS_REGISTRATION  \
0                    0.035792      -22766         365243           -12271.0   
1                    0.028663      -11392          -3528            -4800.0   
2                    0.008866      -23709         365243             -465.0   
3                    0.009630      -20636         365243            -7309.0   

   DAYS_ID_PUBLISH  OWN_CAR_AGE  FLAG_MOBIL  FLAG_EMP_PHONE  FLAG_WORK_PHONE  \
0             -385          NaN           1               0                0   
1            -3920         18.0           1               1                1   
2            -3059         23.0           1               0                0   
3             -712          NaN           1               0                0   

   FLAG_CONT_MOBILE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  CNT_FAM_MEMBERS  \
0                 1           0           0             NaN              2.0   
1                 1           1           0        Managers              4.0   
2                 1           0           0             NaN              2.0   
3                 1           0           0             NaN              1.0   

   REGION_RATING_CLIENT  REGION_RATING_CLIENT_W_CITY  \
0                     2                            2   
1                     2                            2   
2                     2                            2   
3                     2                            2   

  WEEKDAY_APPR_PROCESS_START  HOUR_APPR_PROCESS_START  \
0                     FRIDAY                       11   
1                    TUESDAY                       10   
2                  WEDNESDAY                       20   
3                    TUESDAY                       13   

   REG_REGION_NOT_LIVE_REGION  REG_REGION_NOT_WORK_REGION  \
0                           0                           0   
1                           0                           0   
2                           0                           0   
3                           0                           0   

   LIVE_REGION_NOT_WORK_REGION  REG_CITY_NOT_LIVE_CITY  \
0                            0                       0   
1                            0                       0   
2                            0                       0   
3                            0                       0   

   REG_CITY_NOT_WORK_CITY  LIVE_CITY_NOT_WORK_CITY ORGANIZATION_TYPE  \
0                       0                        0               XNA   
1                       1                        1             Other   
2               

In [ ]:
final_df.shape

(4, 151)

In [ ]:
#Experiment for merging the data
a=df.iloc[:2,:]
b=df_p[df_p["SK_ID_CURR"]==100003]
c=pd.merge(a,b,on="SK_ID_CURR")
display(a,b,c)

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0      100002       1         Cash loans           M            N   
1      100003       0         Cash loans           F            N   

  FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0               Y             0          202500.0    406597.5      24700.5   
1               N             0          270000.0   1293502.5      35698.5   

   AMT_GOODS_PRICE NAME_TYPE_SUITE NAME_INCOME_TYPE  \
0         351000.0   Unaccompanied          Working   
1        1129500.0          Family    State servant   

             NAME_EDUCATION_TYPE    NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  \
0  Secondary / secondary special  Single / not married  House / apartment   
1               Higher education               Married  House / apartment   

   REGION_POPULATION_RELATIVE  DAYS_BIRTH  DAYS_EMPLOYED  DAYS_REGISTRATION  \
0                    0.018801       -9461           -637            -3648.0   
1                    0.003541      -16765          -1188            -1186.0   

   DAYS_ID_PUBLISH  OWN_CAR_AGE  FLAG_MOBIL  FLAG_EMP_PHONE  FLAG_WORK_PHONE  \
0            -2120          NaN           1               1                0   
1             -291          NaN           1               1                0   

   FLAG_CONT_MOBILE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  CNT_FAM_MEMBERS  \
0                 1           1           0        Laborers              1.0   
1                 1           1           0      Core staff              2.0   

   REGION_RATING_CLIENT  REGION_RATING_CLIENT_W_CITY  \
0                     2                            2   
1                     1                            1   

  WEEKDAY_APPR_PROCESS_START  HOUR_APPR_PROCESS_START  \
0                  WEDNESDAY                       10   
1                     MONDAY                       11   

   REG_REGION_NOT_LIVE_REGION  REG_REGION_NOT_WORK_REGION  \
0                           0                           0   
1                           0                           0   

   LIVE_REGION_NOT_WORK_REGION  REG_CITY_NOT_LIVE_CITY  \
0                            0                       0   
1                            0                       0   

   REG_CITY_NOT_WORK_CITY  LIVE_CITY_NOT_WORK_CITY       ORGANIZATION_TYPE  \
0                       0                        0  Business Entity Type 3   
1                       0                        0                  School   

   EXT_SOURCE_1  EXT_SOURCE_2  EXT_SOURCE_3  APARTMENTS_AVG  BASEMENTAREA_AVG  \
0      0.083037      0.262949      0.139376          0.0247            0.0369   
1      0.311267      0.622246           NaN          0.0959            0.0529   

   YEARS_BEGINEXPLUATATION_AVG  YEARS_BUILD_AVG  COMMONAREA_AVG  \
0                       0.9722           0.6192          0.0143   
1                       0.9851           0.7960          0.0605   

   ELEVATORS_AVG  ENTRANCES_AVG  FLOORSMAX_AVG  FLOORSMIN_AVG  LANDAREA_AVG  \
0           0.00         0.0690         0.0833         0.1250        0.0369   
1           0.08         0.0345         0.2917         0.3333        0.0130   

   LIVINGAPARTMENTS_AVG  LIVINGAREA_AVG  NONLIVINGAPARTMENTS_AVG  \
0                0.0202          0.0190                   0.0000   
1                0.0773          0.0549                   0.0039   

   NONLIVINGAREA_AVG  APARTMENTS_MODE  BASEMENTAREA_MODE  \
0             0.0000           0.0252             0.0383   
1             0.0098           0.0924             0.0538   

   YEARS_BEGINEXPLUATATION_MODE  YEARS_BUILD_MODE  COMMONAREA_MODE  \
0                        0.9722            0.6341           0.0144   
1                        0.9851            0.8040           0.0497   

   ELEVATORS_MODE  ENTRANCES_MODE  FLOORSMAX_MODE  FLOORSMIN_MODE  \
0          0.0000          0.0690          0.0833          0.1250   
1          0.0806          0.0345          0.2917          0.3333   

   LANDAREA_MODE  

SK_ID_PREV  SK_ID_CURR NAME_CONTRACT_TYPE  AMT_ANNUITY  \
575941      1810518      100003         Cash loans    98356.995   
1021650     2636178      100003     Consumer loans    64567.665   
1223745     2396755      100003     Consumer loans     6737.310   

         AMT_APPLICATION  AMT_CREDIT  AMT_DOWN_PAYMENT  AMT_GOODS_PRICE  \
575941          900000.0   1035882.0               NaN         900000.0   
1021650         337500.0    348637.5               0.0         337500.0   
1223745          68809.5     68053.5            6885.0          68809.5   

        WEEKDAY_APPR_PROCESS_START  HOUR_APPR_PROCESS_START  \
575941                      FRIDAY                       12   
1021650                     SUNDAY                       17   
1223745                   SATURDAY                       15   

        FLAG_LAST_APPL_PER_CONTRACT  NFLAG_LAST_APPL_IN_DAY  \
575941                            Y                       1   
1021650                           Y                       1   
1223745                           Y                       1   

         RATE_DOWN_PAYMENT  RATE_INTEREST_PRIMARY  RATE_INTEREST_PRIVILEGED  \
575941                 NaN                    NaN                       NaN   
1021650           0.000000                    NaN                       NaN   
1223745           0.100061                    NaN                       NaN   

        NAME_CASH_LOAN_PURPOSE NAME_CONTRACT_STATUS  DAYS_DECISION  \
575941                     XNA             Approved           -746   
1021650                    XAP             Approved           -828   
1223745                    XAP             Approved          -2341   

             NAME_PAYMENT_TYPE CODE_REJECT_REASON NAME_TYPE_SUITE  \
575941                     XNA                XAP   Unaccompanied   
1021650  Cash through the bank                XAP          Family   
1223745  Cash through the bank                XAP          Family   

        NAME_CLIENT_TYPE   NAME_GOODS_CATEGORY NAME_PORTFOLIO  \
575941          Repeater                   XNA           Cash   
1021650        Refreshed             Furniture            POS   
1223745        Refreshed  Consumer Electronics            POS   

        NAME_PRODUCT_TYPE             CHANNEL_TYPE  SELLERPLACE_AREA  \
575941             x-sell  Credit and cash offices                -1   
1021650               XNA                    Stone              1400   
1223745               XNA             Country-wide               200   

         NAME_SELLER_INDUSTRY  CNT_PAYMENT NAME_YIELD_GROUP  \
575941                    XNA         12.0       low_normal   
1021650             Furniture          6.0           middle   
1223745  Consumer electronics         12.0           middle   

                 PRODUCT_COMBINATION  DAYS_FIRST_DRAWING  DAYS_FIRST_DUE  \
575941              Cash X-Sell: low            365243.0          -716.0   
1021650   POS industry with interest            365243.0          -797.0   
1223745  POS household with interest            365243.0         -2310.0   

         DAYS_LAST_DUE_1ST_VERSION  DAYS_LAST_DUE  DAYS_TERMINATION  \
575941                      -386.0         -536.0            -527.0   
1021650                     -647.0         -647.0            -639.0   
1223745                    -1980.0        -1980.0           -1976.0   

         NFLAG_INSURED_ON_APPROVAL  
575941                         1.0  
1021650                        0.0  
1223745                        1.0

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE_x CODE_GENDER FLAG_OWN_CAR  \
0      100003       0           Cash loans           F            N   
1      100003       0           Cash loans           F            N   
2      100003       0           Cash loans           F            N   

  FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT_x  \
0               N             0          270000.0     1293502.5   
1               N             0          270000.0     1293502.5   
2               N             0          270000.0     1293502.5   

   AMT_ANNUITY_x  AMT_GOODS_PRICE_x NAME_TYPE_SUITE_x NAME_INCOME_TYPE  \
0        35698.5          1129500.0            Family    State servant   
1        35698.5          1129500.0            Family    State servant   
2        35698.5          1129500.0            Family    State servant   

  NAME_EDUCATION_TYPE NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  \
0    Higher education            Married  House / apartment   
1    Higher education            Married  House / apartment   
2    Higher education            Married  House / apartment   

   REGION_POPULATION_RELATIVE  DAYS_BIRTH  DAYS_EMPLOYED  DAYS_REGISTRATION  \
0                    0.003541      -16765          -1188            -1186.0   
1                    0.003541      -16765          -1188            -1186.0   
2                    0.003541      -16765          -1188            -1186.0   

   DAYS_ID_PUBLISH  OWN_CAR_AGE  FLAG_MOBIL  FLAG_EMP_PHONE  FLAG_WORK_PHONE  \
0             -291          NaN           1               1                0   
1             -291          NaN           1               1                0   
2             -291          NaN           1               1                0   

   FLAG_CONT_MOBILE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  CNT_FAM_MEMBERS  \
0                 1           1           0      Core staff              2.0   
1                 1           1           0      Core staff              2.0   
2                 1           1           0      Core staff              2.0   

   REGION_RATING_CLIENT  REGION_RATING_CLIENT_W_CITY  \
0                     1                            1   
1                     1                            1   
2                     1                            1   

  WEEKDAY_APPR_PROCESS_START_x  HOUR_APPR_PROCESS_START_x  \
0                       MONDAY                         11   
1                       MONDAY                         11   
2                       MONDAY                         11   

   REG_REGION_NOT_LIVE_REGION  REG_REGION_NOT_WORK_REGION  \
0                           0                           0   
1                           0                           0   
2                           0                           0   

   LIVE_REGION_NOT_WORK_REGION  REG_CITY_NOT_LIVE_CITY  \
0                            0                       0   
1                            0                       0   
2                            0                       0   

   REG_CITY_NOT_WORK_CITY  LIVE_CITY_NOT_WORK_CITY ORGANIZATION_TYPE  \
0                       0                        0            School   
1                       0                        0            School   
2                       0                        0            School   

   EXT_SOURCE_1  EXT_SOURCE_2  EXT_SOURCE_3  APARTMENTS_AVG  BASEMENTAREA_AVG  \
0      0.311267      0.622246           NaN          0.0959            0.0529   
1      0.311267      0.622246           NaN          0.0959            0.0529   
2      0.311267      0.622246           NaN          0.0959            0.0529   

   YEARS_BEGINEXPLUATATION_AVG  YEARS_BUILD_AVG  COMMONAREA_AVG  \
0                       0.9851            0.796          0.0605   
1                       0.9851            0.796          0.0605   
2                       0.9851            0.796          0.0605   

   ELEVATORS_AVG  ENTRANCES_AVG  FLOORSMAX_AVG  FLOORSMIN_AVG  LANDAREA_AVG  \
0           0.08         0.034

In [ ]:
#merging the two dataset
final_df=pd.merge(df, df_p,on='SK_ID_CURR')

In [ ]:
final_df.head()

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE_x CODE_GENDER FLAG_OWN_CAR  \
0      100002       1           Cash loans           M            N   
1      100003       0           Cash loans           F            N   
2      100003       0           Cash loans           F            N   
3      100003       0           Cash loans           F            N   
4      100004       0      Revolving loans           M            Y   

  FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT_x  \
0               Y             0          202500.0      406597.5   
1               N             0          270000.0     1293502.5   
2               N             0          270000.0     1293502.5   
3               N             0          270000.0     1293502.5   
4               Y             0           67500.0      135000.0   

   AMT_ANNUITY_x  AMT_GOODS_PRICE_x NAME_TYPE_SUITE_x NAME_INCOME_TYPE  \
0        24700.5           351000.0     Unaccompanied          Working   
1        35698.5          1129500.0            Family    State servant   
2        35698.5          1129500.0            Family    State servant   
3        35698.5          1129500.0            Family    State servant   
4         6750.0           135000.0     Unaccompanied          Working   

             NAME_EDUCATION_TYPE    NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  \
0  Secondary / secondary special  Single / not married  House / apartment   
1               Higher education               Married  House / apartment   
2               Higher education               Married  House / apartment   
3               Higher education               Married  House / apartment   
4  Secondary / secondary special  Single / not married  House / apartment   

   REGION_POPULATION_RELATIVE  DAYS_BIRTH  DAYS_EMPLOYED  DAYS_REGISTRATION  \
0                    0.018801       -9461           -637            -3648.0   
1                    0.003541      -16765          -1188            -1186.0   
2                    0.003541      -16765          -1188            -1186.0   
3                    0.003541      -16765          -1188            -1186.0   
4                    0.010032      -19046           -225            -4260.0   

   DAYS_ID_PUBLISH  OWN_CAR_AGE  FLAG_MOBIL  FLAG_EMP_PHONE  FLAG_WORK_PHONE  \
0            -2120          NaN           1               1                0   
1             -291          NaN           1               1                0   
2             -291          NaN           1               1                0   
3             -291          NaN           1               1                0   
4            -2531         26.0           1               1                1   

   FLAG_CONT_MOBILE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  CNT_FAM_MEMBERS  \
0                 1           1           0        Laborers              1.0   
1                 1           1           0      Core staff              2.0   
2                 1           1           0      Core staff              2.0   
3                 1           1           0      Core staff              2.0   
4                 1           1           0        Laborers              1.0   

   REGION_RATING_CLIENT  REGION_RATING_CLIENT_W_CITY  \
0                     2                            2   
1                     1                            1   
2                     1                            1   
3                     1                            1   
4                     2                            2   

  WEEKDAY_APPR_PROCESS_START_x  HOUR_APPR_PROCESS_START_x  \
0                    WEDNESDAY                         10   
1                       MONDAY                         11   
2                       MONDAY                         11   
3                       MONDAY                         11   
4                       MONDAY                          9   

   REG_REGION_NOT_LIVE_REGION  REG_REGION_NOT_WORK_REGION  \
0                           0                           0   
1 

In [ ]:
final_df.shape

(1413701, 158)

In [ ]:
lst=list(final_df.columns)

In [ ]:
lst.index("SK_ID_PREV")

122

In [ ]:
col_lst=[]
col_lst.append(lst[122])
col_lst.extend(lst[:122])
col_lst.extend(lst[123:])
col_lst

['SK_ID_PREV',
 'SK_ID_CURR',
 'TARGET',
 'NAME_CONTRACT_TYPE_x',
 'CODE_GENDER',
 'FLAG_OWN_CAR',
 'FLAG_OWN_REALTY',
 'CNT_CHILDREN',
 'AMT_INCOME_TOTAL',
 'AMT_CREDIT_x',
 'AMT_ANNUITY_x',
 'AMT_GOODS_PRICE_x',
 'NAME_TYPE_SUITE_x',
 'NAME_INCOME_TYPE',
 'NAME_EDUCATION_TYPE',
 'NAME_FAMILY_STATUS',
 'NAME_HOUSING_TYPE',
 'REGION_POPULATION_RELATIVE',
 'DAYS_BIRTH',
 'DAYS_EMPLOYED',
 'DAYS_REGISTRATION',
 'DAYS_ID_PUBLISH',
 'OWN_CAR_AGE',
 'FLAG_MOBIL',
 'FLAG_EMP_PHONE',
 'FLAG_WORK_PHONE',
 'FLAG_CONT_MOBILE',
 'FLAG_PHONE',
 'FLAG_EMAIL',
 'OCCUPATION_TYPE',
 'CNT_FAM_MEMBERS',
 'REGION_RATING_CLIENT',
 'REGION_RATING_CLIENT_W_CITY',
 'WEEKDAY_APPR_PROCESS_START_x',
 'HOUR_APPR_PROCESS_START_x',
 'REG_REGION_NOT_LIVE_REGION',
 'REG_REGION_NOT_WORK_REGION',
 'LIVE_REGION_NOT_WORK_REGION',
 'REG_CITY_NOT_LIVE_CITY',
 'REG_CITY_NOT_WORK_CITY',
 'LIVE_CITY_NOT_WORK_CITY',
 'ORGANIZATION_TYPE',
 'EXT_SOURCE_1',
 'EXT_SOURCE_2',
 'EXT_SOURCE_3',
 'APARTMENTS_AVG',
 'BASEMENTAREA_AVG'

In [ ]:
df=final_df[col_lst]
df.head()

SK_ID_PREV  SK_ID_CURR  TARGET NAME_CONTRACT_TYPE_x CODE_GENDER  \
0     1038818      100002       1           Cash loans           M   
1     1810518      100003       0           Cash loans           F   
2     2636178      100003       0           Cash loans           F   
3     2396755      100003       0           Cash loans           F   
4     1564014      100004       0      Revolving loans           M   

  FLAG_OWN_CAR FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT_x  \
0            N               Y             0          202500.0      406597.5   
1            N               N             0          270000.0     1293502.5   
2            N               N             0          270000.0     1293502.5   
3            N               N             0          270000.0     1293502.5   
4            Y               Y             0           67500.0      135000.0   

   AMT_ANNUITY_x  AMT_GOODS_PRICE_x NAME_TYPE_SUITE_x NAME_INCOME_TYPE  \
0        24700.5           351000.0     Unaccompanied          Working   
1        35698.5          1129500.0            Family    State servant   
2        35698.5          1129500.0            Family    State servant   
3        35698.5          1129500.0            Family    State servant   
4         6750.0           135000.0     Unaccompanied          Working   

             NAME_EDUCATION_TYPE    NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  \
0  Secondary / secondary special  Single / not married  House / apartment   
1               Higher education               Married  House / apartment   
2               Higher education               Married  House / apartment   
3               Higher education               Married  House / apartment   
4  Secondary / secondary special  Single / not married  House / apartment   

   REGION_POPULATION_RELATIVE  DAYS_BIRTH  DAYS_EMPLOYED  DAYS_REGISTRATION  \
0                    0.018801       -9461           -637            -3648.0   
1                    0.003541      -16765          -1188            -1186.0   
2                    0.003541      -16765          -1188            -1186.0   
3                    0.003541      -16765          -1188            -1186.0   
4                    0.010032      -19046           -225            -4260.0   

   DAYS_ID_PUBLISH  OWN_CAR_AGE  FLAG_MOBIL  FLAG_EMP_PHONE  FLAG_WORK_PHONE  \
0            -2120          NaN           1               1                0   
1             -291          NaN           1               1                0   
2             -291          NaN           1               1                0   
3             -291          NaN           1               1                0   
4            -2531         26.0           1               1                1   

   FLAG_CONT_MOBILE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  CNT_FAM_MEMBERS  \
0                 1           1           0        Laborers              1.0   
1                 1           1           0      Core staff              2.0   
2                 1           1           0      Core staff              2.0   
3                 1           1           0      Core staff              2.0   
4                 1           1           0        Laborers              1.0   

   REGION_RATING_CLIENT  REGION_RATING_CLIENT_W_CITY  \
0                     2                            2   
1                     1                            1   
2                     1                            1   
3                     1                            1   
4                     2                            2   

  WEEKDAY_APPR_PROCESS_START_x  HOUR_APPR_PROCESS_START_x  \
0                    WEDNESDAY                         10   
1                       MONDAY                         11   
2                       MONDAY                         11   
3                       MONDAY                         11   
4                       MONDAY                          9   

   REG_REGION_NOT_LIVE_REGION  REG_REGION_NOT_WORK_R

In [ ]:
df.shape

(1413701, 158)

We have 1413701 rows and 158 columns after merging the data from application.csv and prev_application.

In [ ]:
df["TARGET"].value_counts()

0    1291341
1     122360
Name: TARGET, dtype: int64

The dataset seems to be disbalanced with respective to target variable.

In [ ]:
#columns with more than 50% nan values
print("columns with more than 50 % of nan values")
nans=((df.isna().sum()/len(df))*100).sort_values()
print("count: ",nans[nans.values>50].count())
nans[nans.values>50]

columns with more than 50 % of nan values
count:  38


APARTMENTS_MODE             50.364115
APARTMENTS_AVG              50.364115
APARTMENTS_MEDI             50.364115
WALLSMATERIAL_MODE          50.462297
ELEVATORS_AVG               52.996779
ELEVATORS_MODE              52.996779
ELEVATORS_MEDI              52.996779
AMT_DOWN_PAYMENT            53.019698
RATE_DOWN_PAYMENT           53.019698
NONLIVINGAREA_AVG           54.860115
NONLIVINGAREA_MEDI          54.860115
NONLIVINGAREA_MODE          54.860115
EXT_SOURCE_1                56.366870
BASEMENTAREA_AVG            58.308511
BASEMENTAREA_MEDI           58.308511
BASEMENTAREA_MODE           58.308511
LANDAREA_MEDI               59.218533
LANDAREA_MODE               59.218533
LANDAREA_AVG                59.218533
OWN_CAR_AGE                 66.294499
YEARS_BUILD_MODE            66.298107
YEARS_BUILD_AVG             66.298107
YEARS_BUILD_MEDI            66.298107
FLOORSMIN_AVG               67.670604
FLOORSMIN_MODE              67.670604
FLOORSMIN_MEDI              67.670604
LIVINGAPARTM


38 columns with  more than 50%  nan values are dependent on each other in which Rate_interest_Privileged and Rate_interest_Primay with 99% nan value 

In [ ]:
df.describe().transpose()

count          mean            std  \
SK_ID_PREV                    1413701.0  1.922744e+06  532715.335255   
SK_ID_CURR                    1413701.0  2.784813e+05  102811.788043   
TARGET                        1413701.0  8.655296e-02       0.281179   
CNT_CHILDREN                  1413701.0  4.048933e-01       0.717345   
AMT_INCOME_TOTAL              1413701.0  1.733160e+05  198573.372016   
AMT_CREDIT_x                  1413701.0  5.875537e+05  384917.331385   
AMT_ANNUITY_x                 1413608.0  2.701702e+04   13951.164803   
AMT_GOODS_PRICE_x             1412493.0  5.277186e+05  353246.491860   
REGION_POPULATION_RELATIVE    1413701.0  2.074985e-02       0.013347   
DAYS_BIRTH                    1413701.0 -1.632105e+04    4344.557089   
DAYS_EMPLOYED                 1413701.0  6.870203e+04  145277.704778   
DAYS_REGISTRATION             1413701.0 -5.003233e+03    3551.050912   
DAYS_ID_PUBLISH               1413701.0 -3.034563e+03    1507.376375   
OWN_CAR_AGE                    476495.0  1.201195e+01      12.067141   
FLAG_MOBIL                    1413701.0  1.000000e+00       0.000000   
FLAG_EMP_PHONE                1413701.0  8.064393e-01       0.395089   
FLAG_WORK_PHONE               1413701.0  1.822585e-01       0.386058   
FLAG_CONT_MOBILE              1413701.0  9.992452e-01       0.027462   
FLAG_PHONE                    1413701.0  2.828752e-01       0.450396   
FLAG_EMAIL                    1413701.0  7.242904e-02       0.259197   
CNT_FAM_MEMBERS               1413701.0  2.150501e+00       0.900679   
REGION_RATING_CLIENT          1413701.0  2.059638e+00       0.495780   
REGION_RATING_CLIENT_W_CITY   1413701.0  2.037603e+00       0.488643   
HOUR_APPR_PROCESS_START_x     1413701.0  1.198433e+01       3.232181   
REG_REGION_NOT_LIVE_REGION    1413701.0  1.207327e-02       0.109213   
REG_REGION_NOT_WORK_REGION    1413701.0  4.396686e-02       0.205021   
LIVE_REGION_NOT_WORK_REGION   1413701.0  3.604793e-02       0.186410   
REG_CITY_NOT_LIVE_CITY        1413701.0  7.360538e-02       0.261128   
REG_CITY_NOT_WORK_CITY        1413701.0  2.221481e-01       0.415690   
LIVE_CITY_NOT_WORK_CITY       1413701.0  1.740601e-01       0.379161   
EXT_SOURCE_1                   616842.0  5.054990e-01       0.210748   
EXT_SOURCE_2                  1411880.0  5.114275e-01       0.191119   
EXT_SOURCE_3                  1171859.0  4.896986e-01       0.196536   
APARTMENTS_AVG                 701703.0  1.149783e-01       0.104870   
BASEMENTAREA_AVG               589393.0  8.698938e-02       0.080745   
YEARS_BEGINEXPLUATATION_AVG    729019.0  9.772352e-01       0.061647   
YEARS_BUILD_AVG                476444.0  7.496433e-01       0.113048   
COMMONAREA_AVG                 428502.0  4.340559e-02       0.073492   
ELEVATORS_AVG                  664485.0  7.525023e-02       0.130434   
ENTRANCES_AVG                  706919.0  1.487248e-01       0.099005   
FLOORSMAX_AVG                  715193.0  2.223169e-01       0.140943   
FLOORSMIN_AVG                  457041.0  2.287916e-01       0.158112   
LANDAREA_AVG                   576528.0  6.561416e-02       0.079438   
LIVINGAPARTMENTS_AVG           449351.0  9.886160e-02       0.090477   
LIVINGAREA_AVG                 709301.0  1.047002e-01       0.107025   
NONLIVINGAPARTMENTS_AVG        434055.0  8.608387e-03       0.047742   
NONLIVINGAREA_AVG              638143.0  2.747308e-02       0.067045   
APARTMENTS_MODE                701703.0  1.120754e-01       0.104846   
BASEMENTAREA_MODE              589393.0  8.627123e-02       0.082634   
YEARS_BEGINEXPLUATATION_MODE   729019.0  9.765155e-01       0.067386   
YEARS_BUILD_MODE               476444.0  7.570512e-01       0.109991   
COMMONAREA_MODE                428502.0  4.153704e-02       0.072112   
ELEVATORS_MODE                 664485.0  7.115661e-02       0.128482   
ENTRANCES_MODE                 706919.0  1.444221e-01       0.099944   
FLOORSMAX_MODE                 715193.0  2.185555e-01       0.140270   
FL

In [ ]:
#columns which are similar with 2 dataset after merging
import re 
for column in df.columns:
  if re.search(".*_x$",str(column))!=None:
    print(re.search(".*_x$",str(column)).group())



NAME_CONTRACT_TYPE_x
AMT_CREDIT_x
AMT_ANNUITY_x
AMT_GOODS_PRICE_x
NAME_TYPE_SUITE_x
WEEKDAY_APPR_PROCESS_START_x
HOUR_APPR_PROCESS_START_x


# **SUMMARY**

1. We have 1413701 rows and 158 columns after merging the data from application.csv and prev_application.


2. 38 columns with  more than 50%  nan values are dependent on each other in which Rate_interest_Privileged and Rate_interest_Primay with 99% nan value 

3. We have 8% of defaulter i.e with payment issue according to the target variable.

4. In most of the cases we have low standard deviation which will tell us less variance  and values closer to mean which is good for classification

5. The dataset seems to be disbalanced with respective to target variable (92% 8% split.)

6. positively skewed data in most of our columns

7. Columns which are similar after merging
  
  NAME_CONTRACT_TYPE_x
  
  AMT_CREDIT_x

  AMT_ANNUITY_x

  AMT_GOODS_PRICE_x

  NAME_TYPE_SUITE_x

  WEEKDAY_APPR_PROCESS_START_x

  HOUR_APPR_PROCESS_START_x


Create hypothesis if required.


id   preid

1    4,5,8